neurom/core/soma.py

```diff
@@ -149,6 +158,29 @@ def __str__(self):
         return ('SomaCylinders(%s) <center: %s, virtual radius: %s>' %
                 (repr(self.points), self.center, self.radius))
 
+    def overlaps(self, points, exclude_boundary=False):
```

Comment:
hard to see what' going one quickly here, is it like checking if a point is in convex hull or the soma.points, or something like that?

Reply:
In this case each segment represents a cylinder, so we check if each point is inside at least one cylinder.

Reply:
not sure exactly what you did, but I just did this cylinder check a few days ago, you want to do this:
if cylinder is parametrized by pos_vec + x * dir_vec (2 vectors and parameter x) for the centerline and radius r, the distance from a point p to centerline of cylinder is np.linalg.norm(np.cross(pos_vec-p, dir_vec)), which you can just check against the radius. Maybe this is what you did ;)

Reply:
Yes, that basically what I did. I compute the longitudinal and orthogonal components and check if the orthogonal is less than radius and if the longitudinal is within the range of the cylinder axis.